### PR TITLE
Add info on programmatic usage

### PR DESCRIPTION
**Pull request by maneetgoyal**
- Addresses one task from #104 
- Adds info on CLI output path

### CODE REVIEW

**Comment by maneetgoyal** [comment]
Hi @domoritz, please review if you get a chance. Was in need of this info for my project but it was missing. Might be helpful for the rest.

**Comment by maneetgoyal** [comment]
Hi @domoritz, can you review it?

File: README.md [hunk 1]
```diff
@@ -17,11 +17,51 @@ Inspired by [`YousefED/typescript-json-schema`](https://github.com/YousefED/type
...
 This project is made possible by a [community of contributors](https://github.com/vega/ts-json-schema-generator/graphs/contributors). We welcome contributions of any kind (issues, code, documentation, examples, tests,...). Please read our [code of conduct](https://github.com/vega/vega/blob/master/CODE_OF_CONDUCT.md).
 
-## Usage
+## CLI Usage
 
 ```bash
 npm install --save ts-json-schema-generator
-./node_modules/.bin/ts-json-schema-generator --path 'my/project/**.*.ts' --type 'My.Type.Full.Name'
+./node_modules/.bin/ts-json-schema-generator --path 'my/project/**.*.ts' --type 'My.Type.Full.Name'  // Logs the schema to console
+./node_modules/.bin/ts-json-schema-generator --path 'my/project/**.*.ts' --type 'My.Type.Full.Name' --out 'my/output/file.json'
```
Comment (domoritz) [inline README.md]:
```suggestion
./node_modules/.bin/ts-json-schema-generator --path 'my/project/**.*.ts' --type 'My.Type.Full.Name'
```

This is already documented

File: README.md [hunk 2]
```diff
@@ -17,11 +17,51 @@ Inspired by [`YousefED/typescript-json-schema`](https://github.com/YousefED/type
...
+```
+
+## Programmatic Usage
+
+```js
+// ts-json-schema.config.js
```
Comment (domoritz) [inline README.md]: why config.js ?

File: README.md [hunk 3]
```diff
@@ -17,11 +17,51 @@ Inspired by [`YousefED/typescript-json-schema`](https://github.com/YousefED/type
...
+
+const tsj = require("ts-json-schema-generator");
+const fs = require("fs");
+
+// Use default config
+const DEFAULT_CONFIG = tsj.DEFAULT_CONFIG;
```
Comment (domoritz) [inline README.md]: unused variable

File: README.md [hunk 4]
```diff
@@ -17,11 +17,51 @@ Inspired by [`YousefED/typescript-json-schema`](https://github.com/YousefED/type
...
+
+const output_path = "path/to/output/file";
+
+try {
+    const schema = tsj.createGenerator(config).createSchema(config.type);
+    const schemaString = JSON.stringify(schema, null, 2);
+    fs.writeFile(output_path, schemaString, err => {
+        if (err) throw err;
+    });
+} catch (error) {
```
Comment (domoritz) [inline README.md]: Not sure why this error handling is needed. I'd omit it.

File: README.md [hunk 5]
```diff
@@ -17,11 +17,51 @@ Inspired by [`YousefED/typescript-json-schema`](https://github.com/YousefED/type
...
+    if (error instanceof tsj.BaseError) {
+        process.stderr.write(tsj.formatError(error));
+        process.exit(1);
+    } else {
+        throw error;
+    }
+}
+
+// Run the command as `node ts-json-schem.config.js`
```
Comment (domoritz) [inline README.md]: Move outside code snippet

File: README.md (hunk 2 above)
Comment (maneetgoyal) [reply README.md]: - Making it `ts-json-schema.js`. 
- It was some pattern we used for server side scripts in our project, checked in by mistake.

File: README.md (hunk 3 above)
Comment (maneetgoyal) [reply README.md]: - Commenting out that line.
- Wanted to showcase that a default config also exists.

File: README.md (hunk 4 above)
Comment (maneetgoyal) [reply README.md]: Removing it

File: README.md (hunk 5 above)
Comment (maneetgoyal) [reply README.md]: Moving it out

File: README.md (hunk 1 above)
Comment (maneetgoyal) [reply README.md]: - The CLI option was missing from [here](https://github.com/vega/ts-json-schema-generator#options), so added that and showed an example usage above. 
- Can remove `./node_modules/.bin/ts-json-schema-generator --path 'my/project/**.*.ts' --type 'My.Type.Full.Name' ` if thats preferred. 
- `./node_modules/.bin/ts-json-schema-generator --path 'my/project/**.*.ts' --type 'My.Type.Full.Name' --out 'my/output/file.json'` seems more useful than the other command.

Comment (domoritz) [reply README.md]: You can always pipe to a file so I don’t agree.

File: README.md (hunk 2 above)
Comment (domoritz) [reply README.md]: Or just main.js

File: README.md (hunk 3 above)
Comment (domoritz) [reply README.md]: Let’s remove it. We don’t have to document every variable.

File: README.md (hunk 4 above)
Comment (domoritz) [reply README.md]: Thanks

File: README.md (hunk 5 above)
Comment (domoritz) [reply README.md]: Thanks

File: README.md (hunk 1 above)
Comment (maneetgoyal) [reply README.md]: Okay got it, I didn't think of piping.

File: README.md (hunk 2 above)
Comment (maneetgoyal) [reply README.md]: Changing it

File: README.md (hunk 3 above)
Comment (maneetgoyal) [reply README.md]: Removing it

**Comment by domoritz** [comment]
Thank you

**Pull request merged**